test/suites/common/xcm/test-mock-dmp-transact.ts

```diff
@@ -0,0 +1,120 @@
+    let polkadotJs: ApiPromise;
+    let transferredBalance;
+    let sendingAddress;
+    let descendAddress;
+    let alice;
+    let chain;
+
+    beforeAll(async function () {
+        polkadotJs = context.polkadotJs();
+        chain = polkadotJs.consts.system.version.specName.toString();
+        alice = chain == 'frontier-template' ? alith : (new Keyring({ type: 'sr25519' }).addFromUri('//Alice', { name: 'Alice default' }));
+        let descendFunction = chain == 'frontier-template' ? descendParentOriginForAddress20 : descendParentOriginFromAddress32;
+
+        const { originAddress, descendOriginAddress } = descendFunction(context);
+
+        sendingAddress = originAddress;
+        descendAddress = descendOriginAddress;
+
+        transferredBalance = 10_000_000_000_000n;
+
+        const txSigned = polkadotJs.tx.balances.transfer(descendOriginAddress, transferredBalance);
+
+        await expectOk(
+          context.createBlock(
+            await txSigned.signAsync(alice)
+          )
+        );
+        const balanceSigned = (
+          (await polkadotJs.query.system.account(descendOriginAddress)) as any
+        ).data.free.toBigInt();
+        expect(balanceSigned).to.eq(transferredBalance);
+    });
+
+    it({
+      id: "T01",
+      title: "Should succeed using sovereign account from signed origin",
+      test: async function () {
+        // Generate random receiver address
+        let random: KeyringPair;
+        random = chain == 'frontier-template' ? generateKeyringPair() : generateKeyringPair("sr25519");
+        // Get Pallet balances index
+        const metadata = await polkadotJs.rpc.state.getMetadata();
+        const balancesPalletIndex = (metadata.asLatest.toHuman().pallets as Array<any>).find(
+        (pallet) => {
+            return pallet.name === "Balances";
+        }
+        ).index;
+
+        const transferCall = polkadotJs.tx.balances.transfer(
+        random.address,
+        transferredBalance / 10n
+        );
+        const transferCallEncoded = transferCall?.method.toHex();
+
+        // We are going to test that we can receive a transact operation from parachain 1
+        // using descendOrigin first
+        const xcmMessage = new XcmFragment({
+            assets: [
+                {
+                multilocation: {
+                    parents: 0,
+                    interior: {
+                    X1: { PalletInstance: balancesPalletIndex },
+                    },
+                },
+                fungible: transferredBalance / 4n,
+                },
+            ],
+            descend_origin: sendingAddress,
+        })
+        .descend_origin()
+        .withdraw_asset()
+        .buy_execution()
+        .push_any({
+            Transact: {
+            originType: "SovereignAccount",
+            requireWeightAtMost: new BN(1000000000),
+            call: {
+                encoded: transferCallEncoded,
+            },
+            },
+        })
+        .as_v2();
```

Comment:
I think it should be better to build these tests using xcm V3 from the beginning.

Reply:
I know but I did not have your PR yet :), I might do it now